test/iron-overlay-behavior.html

```diff
@@ -155,6 +155,21 @@
           assert.equal(getComputedStyle(overlay).display, 'none', 'overlay starts hidden');
         });
 
+        test('_renderOpened called only after is attached', function(done) {
+          var overlay = document.createElement('test-overlay');
+          // overlay is ready at this point, but not yet attached.
+          var spy = sinon.spy(overlay, '_renderOpened');
+          // this triggers _renderOpened
+          overlay.opened = true;
+          // Even if not attached yet, overlay should be the current overlay!
+          assert.equal(overlay, overlay._manager.currentOverlay(), 'currentOverlay ok');
+          // wait for requestAnimationFrame to kick.
+          overlay.async(function() {
```

Comment:
Here I'm waiting "long enough" to see what that `opened` change has triggered. I don't see another way to check for this, either increase the time or use `requestAnimationFrame`..I prefer the timeout of 100ms to the latter.
PS: when I run this test without my fix I get it failing.